### ISSUE TITLE
discussion(connector): persist avro/proto definition in meta

### ISSUE THREAD

**Issue by xiangjinwu** [risingwavelabs/risingwave#12982]
Do we still choose to fetch the descriptor from registry every time we create a new formatter instead of persisting it in meta when creating the sink?

_Originally posted by @wenym1 in https://github.com/risingwavelabs/risingwave/pull/12858#discussion_r1366954735_

**Comment by hzxa21** [comment]
Just to confirm, the benefit of persisting avro/proto definition in meta is not needing to fetch it from schema registry on restart/recovery?

**Comment by wenym1** [comment]
> Just to confirm, the benefit of persisting avro/proto definition in meta is not needing to fetch it from schema registry on restart/recovery?

It's also for consistency and stability. If we don't persisting the metadata in meta, if the value is changed in the external system, RW may be implicitly affected.

**Comment by tabVersion** [comment]
> > Just to confirm, the benefit of persisting avro/proto definition in meta is not needing to fetch it from schema registry on restart/recovery?
> 
> It's also for consistency and stability. If we don't persisting the metadata in meta, if the value is changed in the external system, RW may be implicitly affected.

Yes, but maybe we just need to persist the version number instead of the whole descriptor set.

**Comment by fuyufjh** [comment]
By the way, correspondingly, we shall provide a command `ALTER TABLE/SOURCE ... REFRESH SCHEMA`  so that users can update the schema easily. 

Related #15025. After that, `REFRESH SCHEMA` can be considered as a simpler shortcut for altering the schema configs but change nothing.

**Comment by st1page** [comment]
When executing `CREATE SINK INTO TABLE`, the SQL statement must be fully processed again to generate a plan. https://github.com/risingwavelabs/risingwave/blob/9d6594e6a42f64f8f1003a0783ecb0c586c3427c/src/frontend/src/handler/create_sink.rs#L449 In this case, it will also fetch the external defineiton again which could be invaild.

**Comment by xxchan** [comment]
I elaborated what problems may happen here:

https://github.com/risingwavelabs/risingwave/pull/18419/files#r1753603090

(And this PR fixed a bug caused by the exact problem
